File: 2210-CountHillsandValleysinanArray/2210-CountHillsandValleysinanArray.py
# Last updated: 7/27/2025, 4:32:37 PM
class Solution:
    def countHillValley(self, nums: List[int]) -> int:
        res = 0
        cleaned = [nums[0]]
        for num in nums[1:]:
            if num != cleaned[-1]:
                cleaned.append(num)

        for i in range(1, len(cleaned) - 1):
            if cleaned[i] > cleaned[i - 1] and cleaned[i] > cleaned[i + 1]:
                res += 1
            elif cleaned[i] < cleaned[i - 1] and cleaned[i] < cleaned[i + 1]:
                res += 1



        return res

File: 2210-CountHillsandValleysinanArray/2210-CountHillsandValleysinanArray.cpp
// Last updated: 7/27/2025, 5:15:19 PM
class Solution {
public:
    int countHillValley(vector<int>& nums) {
        vector<int> cleaned;
        cleaned.push_back(nums[0]);

        for (int i = 1; i < nums.size(); ++i ) {
            if (nums[i] != cleaned.back()) {
                cleaned.push_back(nums[i]);
            }
        }

        int res = 0;

        for (int i = 1; i < cleaned.size() - 1; ++i) {
            if ( (cleaned[i] < cleaned[i - 1] && cleaned[i] < cleaned[i + 1]) || (cleaned[i] > cleaned[i - 1] && cleaned[i] > cleaned[i + 1])) {
                res += 1;
            }
        }

        return res;
    }
};